internal/copy_handler.go

```diff
@@ -55,87 +101,49 @@ func getObjectsToCopy(backupName string, from storage.Folder, to storage.Folder,
 	return infos, nil
 }
 
-func copyBackup(backup *Backup, from storage.Folder, to storage.Folder) ([]copyingInfo, error) {
+func GetBackupObjects(backup *Backup, from storage.Folder, to storage.Folder) ([]CopyingInfo, error) {
 	tracelog.InfoLogger.Print("Collecting backup files...")
 	var backupPrefix = path.Join(utility.BaseBackupPath, backup.Name)
 	var objects, err = storage.ListFolderRecursively(from)
 	if err != nil {
 		return nil, err
 	}
 	var hasBackupPrefix = func(object storage.Object) bool { return strings.HasPrefix(object.GetName(), backupPrefix) }
-	return buildCopyingInfos(from, to, objects, hasBackupPrefix), nil
+	return BuildCopyingInfos(from, to, objects, hasBackupPrefix), nil
 }
 
-func copyHistory(backup *Backup, from storage.Folder, to storage.Folder) ([]copyingInfo, error) {
+func GetHistoryObjects(backup *Backup, from storage.Folder, to storage.Folder) ([]CopyingInfo, error) {
 	tracelog.InfoLogger.Print("Collecting history files... ")
 	var fromWalFolder = from.GetSubFolder(utility.WalPath)
+	tracelog.InfoLogger.Print("getSubFolder succeess!")
 	var lastWalFilename, err = getLastWalFilename(backup)
 	if err != nil {
-		return nil, err
+		return make([]CopyingInfo, 0), nil
```

Comment:
WTF? Diff shows me that previously you return an error here. But now you just skip it

Reply:
Yes. That's ok when there are no wal files (maybe yet). That just means that i have nothing to copy.